gsva/gsva.xml

```diff
@@ -52,10 +52,22 @@
             gsva_results = gsva(as.matrix(gene_expression_values), gene_sets, min.sz = $min_size_set, max.sz = $max_size_set, kcdf = "$kernel", mx.diff = ${enrichment_statistic.max_diff}, abs.ranking = ${enrichment_statistic.absolute_ranking}, method = "${method.score}", tau = ${method.tau}, ssgsea.norm = ${method.ssgsea_norm}, verbose = TRUE, parallel.sz = num_of_threads)
 
             ## Prepare GSVA output.
-            gsva_results = melt(gsva_results)
-            colnames(gsva_results) = c("Pathway", "Sample", "EnrichmentScore")
+            gsva_melt = melt(gsva_results)
+            colnames(gsva_melt) = c("Pathway", "Sample", "EnrichmentScore")
 
-            write.table(gsva_results, file = "$gsva_out", col.names = TRUE, row.names = FALSE, quote = FALSE, sep = "\t")
+            write.table(gsva_melt, file = "$gsva_out", col.names = TRUE, row.names = FALSE, quote = FALSE, sep = "\t")
+
+            ## Peform the data visualizations
+            source("${__tool_directory__}/table_visualization.R")
+            options(repr.plot.width=15, repr.plot.height=10) 
+            #if $viz.box_plots.do == "do"
+                box = make.box(gsva_melt, "$viz.box_plots.sample", feature.name="Pathway", value.name="EnrichmentScore")
+                ggsave("gsva_box_plot.png", box)
+            #end if
+            #if $viz.heatmap.do == "do"
+            hm = make.heatmap(gsva_results)
+                ggsave("gsva_heatmap.png", hm)
```

Comment:
Does ggsave work for non-ggplot objects?

Reply:
There is a `png` function that can be used instead to save pngs. https://www.rdocumentation.org/packages/grDevices/versions/3.4.1/topics/png

Reply:
Path to save visualization now added to `pheatmap` call